packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -665,15 +665,22 @@ export class Project extends ProjectBase {
       encryptionKey: props.encryptionKey && props.encryptionKey.keyArn,
       badgeEnabled: props.badge,
       cache,
-      name: props.projectName,
+      name: new cdk.Token(() => this.tryPhysicalName).toString(),
       timeoutInMinutes: props.timeout,
       secondarySources: new cdk.Token(() => this.renderSecondarySources()),
       secondaryArtifacts: new cdk.Token(() => this.renderSecondaryArtifacts()),
       triggers: this.source.buildTriggers(),
     });
 
-    this.projectArn = resource.projectArn;
-    this.projectName = resource.ref;
+    this.projectArn = new cdk.CrossAccountRegionPhysicalArnToken(resource.projectArn, new cdk.Token(() => this.node.stack.formatArn({
+      service: 'codebuild',
+      resource: 'project',
+      resourceName: this.physicalName,
+    })), this).toString();
+    this.projectName = new cdk.CrossAccountRegionPhysicalNameToken(resource.projectName, this).toString();
+    if (props.projectName) {
+      this.assignPhysicalNameIfNotSet(props.projectName);
```

Comment:
If you make this function take a `string | undefined`, you won't need to write this `if` in every location, and I don't think it confuses the semantics much.

Reply:
No longer used like that.